src/array.rs

```diff
@@ -533,7 +545,7 @@ impl<T: Element, D: Dimension> PyArray<T, D> {
                 T::get_dtype_bound(py).into_dtype_ptr(),
                 if is_fortran { -1 } else { 0 },
             );
-            Self::from_owned_ptr(py, ptr)
+            Bound::from_owned_ptr(py, ptr).downcast_into_unchecked()
```

Comment:
This build-untyped-then-downcast is necessary because those `FfiPtrExt` extension traits are not public, right?

Reply:
Meaning https://github.com/PyO3/pyo3/issues/3965

Reply:
Yep, the downcast is needed anyway, because `FFiPtrExt` also return a `Bound<PyAny>`, but I agree that the method chaining it allows does look nicer.

Reply:
Ok, so we cannot really do anything about it here.

For me, it is the downcast which I dislike, not the lack of method chaining. We loose type information that we already have only to recreate it via unchecked downcasting.